tests/doc/test-levels.md

```diff
@@ -0,0 +1,150 @@
+
+Maintain a *small* number of integration tests which use simapp directly to start up
+and test application operations.
+
+Integration tests outnumber end to end tests.
+
+### Simulation / generative testing 
+
+Simulations are a special case of integration tests where deterministically random module
+operations are executed against a running simapp.  No assertions are made for the state
+transitions resulting from module operations but any error will halt and fail the
+simulation.
+
+Modules must implement [AppModuleSimulation.WeightedOperations](https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/types/module/simulation.go#L31) to define their
+simulation operations.  Note that not all modules implement this which may indicate a
+gap in current simulation test coverage.
+
+Modules not returning simulation operations: 
+
+- `x/auth`
+- `x/capability`
+- `x/evidence`
+- `x/mint`
+- `x/params`
+
+A separate binary, [runsim](https://github.com/cosmos/tools/tree/master/cmd/runsim), is responsible for kicking off some of these tests and
+managing their life cycle.
+
+### Limitations
+
+- [A success](https://github.com/cosmos/cosmos-sdk/runs/7606931983?check_suite_focus=true) may take a long time to run, 7-10 minutes per simulation in CI. 
+- [Timeouts](https://github.com/cosmos/cosmos-sdk/runs/7606932295?check_suite_focus=true) sometimes occur on apparently successes without any indication why.
+- Useful error messages not provided on [failure](https://github.com/cosmos/cosmos-sdk/runs/7606932548?check_suite_focus=true) from CI, requiring a developer to run
+  the simulation locally to reproduce.
+
+#### Proposal
+
+No changes proposed at this time.
+
+## End to end tests
+
+End to end tests exercise the entire system as we understand it in as close an approximation
+to a production environment as is practical.  Presently these tests are located at
+[tests/e2e](https://github.com/cosmos/cosmos-sdk/tree/main/tests/e2e) and rely on [testutil/network](https://github.com/cosmos/cosmos-sdk/tree/main/testutil/network) to start up an in-process Tendermint node.
+
+#### Limitations
+
+In general the limitations of end to end tests are orchestration and compute cost.
+Scaffolding is required to start up and run a prod-like environment and the this
+process takes much longer to start and run than unit or integration tests.
+
+Global locks present in Tendermint code cause stateful starting/stopping to sometimes hang
+or fail intermittently when run in a CI environment.
+
+The scope of e2e tests has been complected with command line interface testing.
+
+#### Proposal
+
+Transition existing end to end tests integration tests by removing the dependency on the
+test network and in-process tendermint node, ensuring we're not losing test coverage. 
+
+Keep a small number existing end to end tests but transition to a runner powered by Docker
+via [dockertest](https://github.com/ory/dockertest).
```

Comment:
Just a small note that this is probably better for CI env, but slightly more tedious for DX when writing/debugging e2e tests. Ideally it would be configurable, but we can discuss this later down the road.